control-plane/pkg/security/secret.go

```diff
@@ -165,18 +165,20 @@ func sslConfig(protocol string, data map[string][]byte) kafka.ConfigOption {
 			if err != nil {
 				return fmt.Errorf("[protocol %s] %w", protocol, err)
 			}
-
+			userKeyCert, userKeyExists := data[UserKey]
+			userCert, userCertExists := data[UserCertificate]
+			if !skipClientAuth && !userKeyExists && !userCertExists {
+				skipClientAuth = true
+			}
```

Comment:
Instead of doing it in this logic, which would make the user.skip feature useless (and I like that people are explicit on this aspect for Broker/Channel), can we do it when we create the virtual secret in https://github.com/knative-sandbox/eventing-kafka-broker/blob/2bfe0bd0fc02e2b03a52240198a7287ca9958366/control-plane/pkg/security/secrets_provider_net_spec.go#L42-L68 that is specific to KafkaSource